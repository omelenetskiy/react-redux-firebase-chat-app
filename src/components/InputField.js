import React from 'react';
import PropTypes from 'prop-types';

const InputField = ({ type, onChange, value, name, text }) => {
	return (
		<label className="ui-field">
			<input value={value} type={type} name={name} onChange={onChange} />
			<span>{text}</span>
		</label>
	);
};

export default InputField;

InputField.propTypes = {
	type: PropTypes.string,
	onChange: PropTypes.func,
	value: PropTypes.string,
	text: PropTypes.string,
	name: PropTypes.string,
};
