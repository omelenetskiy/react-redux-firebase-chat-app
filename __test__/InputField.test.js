import React from 'react'
import { shallow } from 'enzyme'
import InputField from '../src/components/InputField'

describe('InputField', () => {
  let wrap;
  let props;
  beforeEach(() => {
    props = { type: 'text', onChange: jest.fn(), value: 'Type here...', name: 'text', text: 'Text field' }
    wrap = shallow(<InputField {...props} />)
  })
  it('should render input', () => {
    expect(wrap).toMatchSnapshot();
  })
  it('should render span text', () => {
    expect(wrap.find('span').text()).toEqual('Text field');
  })
  it('should simulate onChange', () => {
    const event = jest.fn();
    wrap.find('input').simulate('change', event);
    expect(props.onChange).toHaveBeenCalledTimes(1);
  })
})
