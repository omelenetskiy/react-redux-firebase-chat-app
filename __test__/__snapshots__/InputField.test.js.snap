// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField should render input 1`] = `
<label
  className="ui-field"
>
  <input
    name="text"
    onChange={[MockFunction]}
    type="text"
    value="Type here..."
  />
  <span>
    Text field
  </span>
</label>
`;
